DelvUI/Interface/NinjaHudWindow.cs

```diff
@@ -46,6 +55,109 @@ protected override void Draw(bool _)
+            float yPos = CenterY + _config.Position.Y + _config.MudraBarPosition.Y - _config.MudraBarSize.Y / 2f;
+
+            BarBuilder builder = BarBuilder.Create(xPos, yPos, _config.MudraBarSize.Y, _config.MudraBarSize.X);
+
+            // each of the 2 mudra charges has a cooldown of 20s
+            float maximum = 40f;
+            // get the current cooldown and number of charges on mudras
+            float mudraCooldownInfo = _spellHelper.GetSpellCooldown(2259);
+            int mudraStacks = _spellHelper.GetStackCount(2, 2259);
+
+            // is the player casting ninjutsu or under kassatsu?
+            IEnumerable<StatusEffect> ninjutsuBuff = PluginInterface.ClientState.LocalPlayer.StatusEffects.Where(o => o.EffectId == 496);
+            IEnumerable<StatusEffect> kassatsuBuff = PluginInterface.ClientState.LocalPlayer.StatusEffects.Where(o => o.EffectId == 497);
+            IEnumerable<StatusEffect> tcjBuff = PluginInterface.ClientState.LocalPlayer.StatusEffects.Where(o => o.EffectId == 1186);
+            bool haveMudraBuff = ninjutsuBuff.Any();
+            bool haveKassatsuBuff = kassatsuBuff.Any();
+            bool haveTCJBuff = tcjBuff.Any();
+            // for some reason (perhaps a slight delay), the mudras may be on cooldown before the "Mudra" buff is applied
+            // hence we check for either
+            bool inNinjutsu = mudraStacks == -2 || haveMudraBuff;
+            // this ensures that if the cooldown suddenly drops to 0.5s because the player has casted a mudra
+            // then the depicted cooldown freezes while the ninjutsu is being casted
+            // unfortunately I can't quite get this to work for kassatsu
+            // this is really only a problem if we wish to keep showing chunking during ninjutsu casts
+            if (inNinjutsu)
+            {
+                mudraCooldownInfo = _oldMudraCooldownInfo;
+            }
+            else
+            {
+                _oldMudraCooldownInfo = mudraCooldownInfo;
+            }
+            if (haveTCJBuff)
+            {
+                PluginLog.Log($"In TCJ: {tcjBuff.First().StackCount}");
+            }
+            // if we are casting ninjutsu then show ninjutsu info
+            // if we are in kassatsu, simply show "kassatsu" unless we are casting ninjutsu
+            // if we are in TCJ, simply show "ten chi jin" unless we are casting ninjutsu (this overrides kassatsu)
+            if (inNinjutsu || haveKassatsuBuff || haveTCJBuff)
+            {
+                string ninjutsuText = haveKassatsuBuff ? "KASSATSU" : "";
+                // determine which ninjutsu is being cast
+                // thanks to daemitus for pointing me in this direction
+                // NOTE: in ClientStructs it seems that StackCount and Param are switched
+                // if this ever breaks -- possibly due to a ClientStructs update -- try swapping them

```

Comment:
Thanks for this;  while I normally dislike comments in code THESE types of comments are great examples of when they are helpful.